api/desecapi/tests/test_rrsets.py

```diff
@@ -239,14 +248,155 @@ def test_create_my_rr_sets_empty_payload(self):
+                with self.assertPdnsRequests(self.requests_desec_rr_sets_update(name=self.my_empty_domain.name)):
+                    response = self.client.post_rr_set(self.my_empty_domain.name, **data)
+                    self.assertStatus(response, status.HTTP_201_CREATED)
+                with self.assertPdnsRequests(self.requests_desec_rr_sets_update(name=self.my_empty_domain.name)):
+                    response = self.client.delete_rr_set(self.my_empty_domain.name, subname='', type_=t)
+                    self.assertStatus(response, status.HTTP_204_NO_CONTENT)
+
+    def test_create_my_rr_sets_known_type_invalid(self):
+        # TODO fill in more examples
+        datas = {
+            # recordtype: [list of examples expected to be rejected, individually]
+            'A': ['127.0.0.999', '127.0.0.256', '::1', 'foobar', '10.0.1', '10!'],
+            'AAAA': ['::g', '1:1:1:1:1:1:1:1:', '1:1:1:1:1:1:1:1:1'],
+            'AFSDB': ['example.com.', '1 1', '1 de'],
+            'CAA': ['43235 issue "letsencrypt.org"'],  # '128 x asdf', '1 issue letsencrypt.org'
+            'CDNSKEY': [],
+            'CDS': [],
+            'CERT': ['6 0 sadfdd=='],
+            'CNAME': ['example.com', '10 example.com.'],
+            'DHCID': [],
+            'DLV': [],
+            'DS': ['-34 13 1 aabbccddeeff'],
+            'EUI48': [],
+            'EUI64': [],
+            'HINFO': ['"ARMv8-A"'],
+            'IPSECKEY': [], 'KEY': [], 'KX': [],
+            'LOC': ['23 12 61.000 N 42 22 48.500 W 65.00m 20.00m 10.00m 10.00m', 'foo', '1.1.1.1'],
+            'MAILA': [], 'MAILB': [], 'MINFO': [], 'MR': [],
+            'MX': ['10 example.com', 'example.com.', '-5 asdf.', '65537 asdf.'],
+            'NAPTR': [],
+            'NS': ['ns1.example.com', '127.0.0.1'],
+            'OPENPGPKEY': [],
+            'PTR': ['"example.com."', '10 *.example.com.'],
+            'RP': ['hostmaster.example.com.', '10 foo.'],
+            'SPF': ['"v=spf1', ''],
+            'SRV': ['0 0 0 0', '100 5061 example.com.'],
+            'SSHFP': ['aabbcceeddff'],
+            'TKEY': [],
+            'TLSA': ['3 1 1 AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA'],
+            'TSIG': [],
+            'TXT': ['foob"ar'],  # 'foobar'
+            'URI': [], 'WKS': []
+        }
+        for t, records in datas.items():
+            for r in records:
+                data = {'records': [r], 'ttl': 3660, 'type': t, 'subname': ''}
+                response = self.client.post_rr_set(self.my_empty_domain.name, **data)
                 self.assertStatus(response, status.HTTP_400_BAD_REQUEST)
 
+    def test_create_my_rr_sets_unknown_type(self):
+        for _type in ['AA', 'ASDF'] + list(RR_SET_TYPES_AUTOMATIC | RR_SET_TYPES_UNSUPPORTED):
+            response = self.client.post_rr_set(self.my_domain.name, records=['1234'], ttl=3660, type=_type)
+            self.assertStatus(response, status.HTTP_400_BAD_REQUEST)
```

Comment:
let's check a bit more specifically that the type was rejected (and not the record content or something else)

Reply:
done